### PR TITLE
Cleanup resources when Shutdown() is called 

### PR DESCRIPTION
**Pull request by acaloiaro**
- Store context cancelFuncs to be called during Shutdown()

### CODE REVIEW

File: memory_backend.go [hunk 1]
```diff
@@ -121,7 +123,13 @@ func (m *MemBackend) Listen(ctx context.Context, queue string, h Handler) (err e
 	m.handlers.Store(queue, h)
 	m.queues.Store(queue, make(chan Job, queueCapacity))
 
-	err = m.start(ctx, queue)
+	cctx, cancel := context.WithCancel(ctx)
```
Comment (jcmuller) [inline memory_backend.go:126]: Probably no need to create a new context var here, just reassign the `ctx`. When would you want to use the original context without the cancellable?

File: postgres_backend.go [hunk 2]
```diff
@@ -48,14 +48,15 @@ const (
...
-	mu         *sync.Mutex         // mutext to protect mutating state on a pgWorker
-	futureJobs map[int64]time.Time // map of future job IDs to their due time
-	logger     Logger
+	cancelFuncs []context.CancelFunc // A collection of cancel functions to be called upon Shutdown()
+	config      *pgConfig
+	cron        *cron.Cron
+	listenConn  *pgx.Conn
+	pool        *pgxpool.Pool
+	handlers    map[string]Handler  // a map of queue names to queue handlers
+	mu          *sync.Mutex         // mutext to protect mutating state on a pgWorker
```
Comment (jcmuller) [inline postgres_backend.go:57]:
```suggestion
	mu          *sync.Mutex         // mutex to protect mutating state on a pgWorker
```

File: postgres_backend.go [hunk 3]
```diff
@@ -82,22 +83,28 @@ type PgBackend struct {
...
-		handlers:   make(map[string]Handler),
-		futureJobs: make(map[int64]time.Time),
-		logger:     slog.New(slog.NewTextHandler(os.Stdout)),
-		cron:       cron.New(),
+	w := &PgBackend{
+		mu:          &sync.Mutex{},
+		config:      &pgConfig{connectString: connectString},
+		handlers:    make(map[string]Handler),
+		futureJobs:  make(map[int64]time.Time),
+		logger:      slog.New(slog.NewTextHandler(os.Stdout)),
```
Comment (jcmuller) [inline postgres_backend.go:91]: Log should be passed in. Maybe a WithSLog option

Comment (acaloiaro) [reply postgres_backend.go:91]: There's a TODO in the code for that already. I'm going to defer on this for now.

**Pull request merged**